[jyztintan] iP

jyztintan opened the pull request:
# TaskYapper built by [Justin Tan](https://www.linkedin.com/in/tan-wee-kian-justin/) 😏 

> "Stop Yappin" - @UdhayaShan1

TaskYapper provides many functionalities including:
- Adding new tasks
- Removing current tasks
- Marking/Unmarking current tasks

To get started, all you need to do is,
1. Download it from [here](https://github.com/jyztintan/ip/releases/tag/A-Jar)
2. Launch a terminal instance at the source location of the Jar file
3. Run `java -jar duke.jar` in the command line
4. Start **yappin**!

Features:
- [x] Managing Tasks
- [x] Managing Deadlines
- [ ]  Reminders (_to be implemented_)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

jyztintan committed:
Rename, Greet, Exit the skeletal version

jyztintan committed:
Adding echo feature

jyztintan committed:
Editing greeting and exiting messages

jyztintan committed:
Fixing minor bug where TaskYapper echoes messages with spaces into multiple lines

jyztintan committed:
Editing greeting and exiting messages

jyztintan committed:
Adding list feature

jyztintan committed:
Changing 'list' keyword to 'yap'

jyztintan committed:
Implement no items message for 'yap' command

jyztintan committed:
Add skeleton for Task class

jyztintan committed:
Refactor: Replace string array with Task array in Duke class

jyztintan committed:
Add mark and unmark checkers in TaskYapper main logic

jyztintan committed:
Add mark and unmark features

jyztintan committed:
Add mark and unmark messages

jyztintan committed:
Refactor code and add checks for already marked/unmarked tasks

jyztintan committed:
Add skeletal code for Event, ToDo and Deadline class

jyztintan committed:
Refactor code for echo feature in TaskYapper main logic

jyztintan committed:
Add new task types - ToDos, Events and Deadlines

jyztintan committed:
Create text-ui-test

jyztintan committed:
Extend TextUITesting

jyztintan committed:
Refactor code to improve readability

jyztintan committed:
Refactor code to improve readability

jyztintan committed:
Fix bug where task reflects its type at the start

jyztintan committed:
Add new DukeException to ensure all commands are valid

jyztintan committed:
Updating test files for new DukeException (ref last commit)

jyztintan committed:
Add appropriate Exception handling

jyztintan committed:
Update error message, make it more clear

jyztintan committed:
Update test expected file to match new behaviour

jyztintan committed:
Change array to use ArrayList instead

jyztintan committed:
Edit TextUITesting

jyztintan committed:
Implement persistent task storage

- Load tasks from 'taskyapper.txt' file at startup
- Save tasks to the file on any modification (add, delete, mark, unmark)
- Ensure creation of file and parent directories if not existing
- Add error handling for file operations

jyztintan committed:
Remove public from all classes

jyztintan committed:
Create new file when data file is corrupted

jyztintan committed:
Merge branch 'branch-Level-7'

jyztintan committed:
Implement Deadline and Event to use LocalDate class instead of Strings

jyztintan committed:
Merge branch 'branch-Level-8'

jyztintan committed:
Change all classes to be public

jyztintan committed:
Refactor code into skeletal OOP Classes

jyztintan committed:
Standardise arguments for execute in enum Commands

jyztintan committed:
Fix bug where trigger messages are still sent for null tasks

jyztintan committed:
Organize the classes into suitable java packages

jyztintan committed:
Merge remote-tracking branch 'origin/add-gradle-support'

jyztintan committed:
Set up gradle to run Duke

jyztintan committed:
Create new JUnit tests

jyztintan committed:
Merge branch 'add-junit-tests'

jyztintan committed:
Update build.gradle

jyztintan committed:
Add Javadocs for Duke.java

jyztintan committed:
Add Javadocs for Ui.java

jyztintan committed:
Add Javadocs for Storage.java

jyztintan committed:
Add Javadocs for Event.java and Deadline.java

jyztintan committed:
Add Javadocs for Todo.java and Task.java

jyztintan committed:
Add Javadocs and refactor TaskList.java

jyztintan committed:
Add Javadocs and refactor DukeException.java

jyztintan committed:
Add Javadocs for Command.java

jyztintan committed:
Add Javadocs for Parser.java

jyztintan committed:
Standardise import statements order

jyztintan committed:
Implement find command

jyztintan committed:
Merge branch 'branch-A-JavaDoc'

jyztintan committed:
Merge branch 'branch-A-CodingStandard'

jyztintan committed:
Merge branch 'branch-Level-9'

jyztintan committed:
Set up Java FX

chiageng commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,71 @@
+ */
 public class Duke {
+
+    private Ui ui;
+    private Storage storage;
+    private TaskList taskList;
```
Good variable naming with camelCase, and it is very intuitive to understand the variable purpose.

chiageng commented on src/main/java/commands/Command.java:
```diff
@@ -0,0 +1,141 @@
+            storage.saveTasks(tasks);
+        }
+    },
+    /**
+     * Command to add a event task to tasklist.
+     */
+    ADD_EVENT {
+        @Override
+        public void execute(TaskList tasks, Ui ui, Storage storage, String message) {
+            // Create a new tasks.Event task and add it to the task list
+            Task task = tasks.initTask(message, "event");
+            tasks.addTasktoTaskList(task);
+            ui.triggerAddMessage(task);
+            storage.saveTasks(tasks);
+        }
+    },
+
+    /**
+     * Command to delete a task from tasklist by index.
+     */
+    DELETE {
+        @Override
+        public void execute(TaskList tasks, Ui ui, Storage storage, String message) {
+            String[] inputs = message.split(" ");
+            int index = Integer.parseInt(inputs[1]);
+            Task task = tasks.removeTaskfromTaskList(index);
+            ui.triggerDeleteMessage(task);
+            storage.saveTasks(tasks);
+        }
+    },
+    FIND {
+        @Override
+        public void execute(TaskList tasks, Ui ui, Storage storage, String message) {
+            String[] inputs = message.split(" ");
+            String queryString = inputs[1];
+            ArrayList<Task> tempList = new ArrayList<>();
+            TaskList tempTaskList = tasks.filter(queryString);
+            tempTaskList.yapTasks();
+        }
+    }
```
It is a very interesting way to handle different Command Type. It is good that all the constant names are uppercase which is very clear to other developer/reviewer.

chiageng commented on src/main/java/exceptions/DukeException.java:
```diff
@@ -0,0 +1,19 @@
+ * Represents an exception specific to the Duke application.
+ * This class is used for handling exceptional scenarios that are unique to the Duke application's workflow.
+ */
+public class DukeException extends Exception{
```
Good coding convention on using PascalCase on class names

chiageng commented on src/main/java/tasks/Task.java:
```diff
@@ -0,0 +1,103 @@
+ */
+public class Task {
+    protected String description; // The description of the task
+    protected boolean isDone; //The completion status of the task
```
Good naming convention to show that it is a boolean variable.

chiageng commented on src/main/java/tasks/Task.java:
```diff
@@ -0,0 +1,103 @@
+     *
+     * @return The task's description.
+     */
+    public String getDescription() {
```
Good naming convention to represent method, it is very intuitive for other developer to understand the purpose of this function.

chiageng reviewed (commented):
In overall, it is a great work alligned with coding standard. Keep it up.

jyztintan committed:
Fix bug where Events cannot be properly stored

jyztintan committed:
Initiate JavaFX skeletal code

jyztintan committed:
Resize test label

jyztintan committed:
Set up skeletal conversation UI

jyztintan committed:
Adding auto-scroll chat user interface

jyztintan committed:
Refactor code into ui and main packages

jyztintan committed:
Add custom behaviour to DialogBox

jyztintan committed:
Add padding between DialogBox, ImageView and Label

jyztintan committed:
Clip image view into circles

jyztintan committed:
Add background color to each dialog box

jyztintan committed:
Use FXML in Duke application

jyztintan committed:
Fully integrate JavaFX GUI

KSSWSept20 commented on src/main/java/commands/Parser.java:
```diff
@@ -0,0 +1,42 @@
+     * @return The Command corresponding to the user's input message.
+     * @throws DukeException If the input message does not match any known command patterns, indicating that the
+     *                       user's instruction is unclear or invalid within the context of the application.
+     */
```
I like the detailed comments which helps me to quickly understand the code.

KSSWSept20 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,71 @@
+            System.out.println(e.getMessage());
+            } catch (IndexOutOfBoundsException e) {
+            System.out.println("Quit yappin, that task does not exist");
+        }
```
I like how clean the code looks due to appropriate indentation.

KSSWSept20 commented on src/main/java/Storage.java:
```diff
@@ -0,0 +1,105 @@
+package main.java;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintWriter;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import tasks.Task;
+import tasks.TaskList;
+import tasks.ToDo;
+import tasks.Deadline;
+import tasks.Event;
```
I like how every imported class is listed explicitly instead of importing all at once using *.

KSSWSept20 commented on src/main/java/Ui.java:
```diff
@@ -0,0 +1,78 @@
+     * Prints a greeting message to the user when the application starts.
+     */
+    public void greet() {
+        String logo = "▀█▀ ▄▀█ █▀ █▄▀ █▄█ ▄▀█ █▀█ █▀█ █▀▀ █▀█\n"
+                + "░█░ █▀█ ▄█ █░█ ░█░ █▀█ █▀▀ █▀▀ ██▄ █▀▄\n";
```
I like the indentation of eight space instead of four, making the long line clean and readable.
By the way, this looks so cool! 🤩

KSSWSept20 reviewed (commented):
Overall, LGTM, I found the code clean and easy to read. 👍 
I really like how every class and function declared comes with detailed comments which ease the readers a lot.
Great job!

jyztintan committed:
Change Greeting message

jyztintan committed:
Add checkstyle

jyztintan committed:
Reformat code to comply with coding standard

jyztintan committed:
Reformat code to comply with coding standard

jyztintan committed:
Add assertions

The current code assumes that empty/invalid
user inputs have been correctly handled,
and should throw an error message back at the user.

Add assertions in the code which ensure
that we have indeed handled such scenario
where user inputs are empty or invalid.

Check, at various points of the code, that the
user input is not null, and the following chain of events
is also valid (for example, Command is also not null).
This enables us to prevent any possible overlooking
of specific edge cases which may not be caught or addressed.

jyztintan committed:
Throw DukeException for invalid inputs in TaskList

jyztintan committed:
Fix bug where Deadline invalid inputs caused crash

jyztintan committed:
Refactor code to organise initialisation of task in TaskList

The previous code for initialisation of task was lengthy and hard to understand.

With these new changes, the initialisation of different types of task
are abstracted out for better readibility and flow. Logic wise, the code is the same.

This way, we are able to change the implementation of a specific types
initialisation without affecting others.

jyztintan committed:
Resolve merge conflict

jyztintan committed:
Merge pull request #2 from jyztintan/A-CodeQuality

 Improve code quality

jyztintan committed:
Merge pull request #3 from jyztintan/add-assertions

Add assertions

jyztintan committed:
Add Schedule Extension

jyztintan committed:
Merge pull request #4 from jyztintan/add-extension

Add Schedule Extension

jyztintan committed:
Refactor code to reduce conditional flow

jyztintan committed:
Add gradle.yml file for CI

jyztintan committed:
Change entry point of application

jyztintan committed:
Fix bug so delete command can now handle IndexOutOfBoundsException

jyztintan committed:
Merge pull request #5 from jyztintan/add-ci

Add Continuous Integration

jyztintan committed:
Implement highlighting of error messages

jyztintan committed:
Add padding between profile pic and text

jyztintan committed:
Add background formatting for messages

jyztintan committed:
Merge pull request #7 from jyztintan/improve-gui

Improve gui

jyztintan committed:
Rename all instances of Duke to TaskYapper

jyztintan committed:
Rename all instances of Duke to TaskYapper

jyztintan committed:
Add error checking for start/end date for events

jyztintan committed:
Merge pull request #8 from jyztintan/add-personality

Add personality

jyztintan committed:
Merge pull request #9 from jyztintan/adding-error-handling

Add error checking

jyztintan committed:
Add and improve Javadocs using ChatGPT

jyztintan committed:
Merge pull request #10 from jyztintan/improve-javadocs

Add and improve Javadocs using ChatGPT

jyztintan committed:
Add representative screenshot

jyztintan committed:
Update screenshot

jyztintan committed:
Add new JUnit tests

jyztintan committed:
Add new JUnit tests

jyztintan committed:
Merge pull request #11 from jyztintan/adding-junit-tests

Adding junit tests

jyztintan committed:
Update README.md

jyztintan committed:
Update README.md

jyztintan committed:
Update README.md

jyztintan committed:
Update README.md

jyztintan committed:
Update README.md

jyztintan committed:
Update User Guide to match proper command formats

jyztintan committed:
Merge pull request #13 from jyztintan/update-user-guide

Update User Guide to match proper command formats

jyztintan committed:
Fix minor bug in User Guide

the command descriptor for adding an event should be `event` not `add_event`

jyztintan committed:
Merge pull request #14 from jyztintan/update-user-guide

Fix minor bug in User Guide

jyztintan committed:
Update overlooked javadoc

Update overlooked javadoc for getResponse method in the TaskYapper.java file

jyztintan committed:
Merge pull request #15 from jyztintan/update-javadoc

Update overlooked javadoc
